File: apps/api/schemas/inscripciones.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

# ============= INSCRIPCION SCHEMAS =============

class InscripcionDetailResponse(BaseModel):
    """Detalle de inscripción con progreso completo"""
    id: str
    progreso: int  # 0-100
    estado: str  # NO_INICIADO, EN_PROGRESO, COMPLETADO, APROBADO, REPROBADO
    inicioDate: Optional[str] = None
    finDate: Optional[str] = None
    cursoId: str
    alumnoId: str
    
    class Config:
        from_attributes = True

class MisCursosItem(BaseModel):
    """Item de curso con estado de inscripción"""
    id: str
    nombre: str
    descripcion: str
    codigo: str
    duracionHoras: int
    progreso: int
    estado: str
    proximaActividad: Optional[str] = None  # Calculado
    
    class Config:
        from_attributes = True

class MisCursosResponse(BaseModel):
    """Respuesta con cursos del usuario y estadísticas"""
    cursos: List[MisCursosItem]
    stats: dict  # { cursosActivos, cursosCompletados, horasAcumuladas }

class InscribirseRequest(BaseModel):
    """Request para inscribirse en un curso"""
    cursoId: str

class CompletarModuloRequest(BaseModel):
    """Request para marcar módulo como completado"""
    moduloId: str
    # Campos opcionales según tipo de módulo
    calificacionQuiz: Optional[float] = None
    aprobadoQuiz: Optional[bool] = None

class CompletarModuloResponse(BaseModel):
    """Respuesta al completar módulo"""
    success: bool
    nuevoProgreso: int
    cursoCompletado: bool
    credencialGenerada: bool
    credencialNumero: Optional[str] = None
    message: str

File: apps/web/lib/api/inscripciones.ts
import {
    MisCursosResponse,
    Inscripcion,
    CompletarModuloResponse,
} from '@/types/training';
import { api } from '../api-client';

export const inscripcionesApi = {
    /**
     * Obtener cursos del usuario actual con progreso
     */
    async misCursos(): Promise<MisCursosResponse> {
        return api.get('/inscripciones/mis-cursos');
    },

    /**
     * Inscribirse en un curso
     */
    async inscribirse(cursoId: string): Promise<Inscripcion> {
        return api.post(`/inscripciones/${cursoId}/inscribir`, {});
    },

    /**
     * Obtener estado de inscripción en un curso
     */
    async obtenerInscripcion(cursoId: string): Promise<Inscripcion> {
        return api.get(`/inscripciones/${cursoId}`);
    },

    /**
     * Completar un módulo
     */
    async completarModulo(
        cursoId: string,
        moduloId: string,
        calificacionQuiz?: number,
        aprobadoQuiz?: boolean
    ): Promise<CompletarModuloResponse> {
        return api.post(
            `/inscripciones/${cursoId}/modulos/${moduloId}/completar`,
            {
                moduloId,
                calificacionQuiz,
                aprobadoQuiz,
            }
        );
    },
};
